esmvaltool/diag_scripts/cos22esd/hotspot_plotter.py

```diff
@@ -0,0 +1,649 @@
+            ]
+        elif dimension == "longitude":
+            boundaries = [
+                region["start_longitude"],
+                region["end_longitude"],
+                region["end_longitude"],
+                region["start_longitude"],
+                region["start_longitude"],
+            ]
+        return boundaries
+
+    def define_projection(self, region):
+        """Projection definition to get LambertConformal borders."""
+        region = {
+            "start_longitude": region[0],
+            "end_longitude": region[1],
+            "start_latitude": region[2],
+            "end_latitude": region[3],
+        }
+        projection = "LambertConformal"
+        plotextend = [
+            region["start_longitude"],
+            region["end_longitude"],
+            region["start_latitude"],
+            region["end_latitude"],
+        ]
+        if projection == "LambertConformal":
+            # plotextend has to be a little larger so everything is on there
+            plotextend = [
+                plotextend[0] - 1.0,
+                plotextend[1] + 1.0,
+                plotextend[2] - 1.0,
+                plotextend[3] + 1.0,
+            ]
+            # path to cut out is exact though
+            lons = self.region_to_square(region, "longitude")
+            lats = self.region_to_square(region, "latitude")
+            path_ext = [[lon, lat] for lon, lat in zip(lons, lats)]
+            path_ext = mpath.Path(path_ext).interpolated(20)
+        # South Hemisfere
+        if region["start_latitude"] <= 0 and region["end_latitude"] <= 0:
+            proj = ccrs.LambertConformal(
+                central_longitude=np.sum(plotextend[:2]) / 2.0,
+                central_latitude=np.sum(plotextend[2:]) / 2.0,
+                cutoff=+30,
+                standard_parallels=(-33, -45),
+            )
+        # North Hemisphere
+        else:
+            proj = ccrs.LambertConformal(
+                central_longitude=np.sum(plotextend[:2]) / 2.0,
+                central_latitude=np.sum(plotextend[2:]) / 2.0,
+            )
+        return proj, projection, path_ext, plotextend
+
+    def sorted_dim(self, cube, coord="longitude"):
+        """Sorts the cube data according to the longitude coordinate values.
+
+        example: 180/-180 --> -180/180
+        """
+        coord_to_sort = cube.coord(coord)
+        assert coord_to_sort.ndim == 1, "Coord should be 1-dimensional."
+        (dim,) = cube.coord_dims(coord_to_sort)
+        index = [slice(None)] * cube.ndim
+        index[dim] = np.argsort(coord_to_sort.points)
+        cube = cube[tuple(index)]
+        coord = cube.coord(coord)
+        coord_dims = cube.coord_dims(coord)
+        # Create a new coordinate which is a DimCoord.
+        dim_coord = iris.coords.DimCoord.from_coord(coord)
+        # Remove the AuxCoord and add the DimCoord.
+        cube.remove_coord(coord_to_sort)
+        cube.add_dim_coord(dim_coord, coord_dims)
```

Comment:
Could something like `iris.util.promote_aux_coord_to_dim_coord(cube, name_or_coord)` help in here?